src/extensions.rs

```diff
@@ -195,4 +195,27 @@ impl<T: AsRef<str>> T {
         let index = s.find(c)?;
         Some((&s[..index], &s[index + c.len_utf8()..]))
     }
+
+    fn split_at_word<'a>(&'a self, split_at: &str) -> (String, String) {
+        let str: &str = self.as_ref();
+        let mut result = (String::from(""), String::from(""));
```

Comment:
No need to have this be mutable, you can just easily return the tuples directly from the code. Also, String::new() exists as a shorter version of String::from("")